File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> hash_map;

        for(string s: strs) {
            string sorted_s = s;
            sort(sorted_s.begin(), sorted_s.end());

            hash_map[sorted_s].push_back(s);
        }

        vector<vector<string>> res;
        for (const auto& pair: hash_map) {
            res.push_back(pair.second);
        }
        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        group = defaultdict(list)

        for s in strs:
            sorted_s = sorted(s)
            sorted_s = ''.join(sorted_s)
            group[sorted_s].append(s)
        return list(group.values())

        
        